homeassistant/components/mqtt/vacuum/schema_legacy.py

```diff
@@ -171,25 +190,40 @@ class MqttVacuum(MqttEntity, VacuumEntity):
     _entity_id_format = ENTITY_ID_FORMAT
     _attributes_extra_blocked = MQTT_LEGACY_VACUUM_ATTRIBUTES_BLOCKED
 
-    def __init__(self, hass, config, config_entry, discovery_data):
+    _encoding: str | None
+    _fan_speed_list: list[str]
+    _qos: bool
+    _retain: bool
+    _payloads: dict[str, str]
+    _send_command_topic: str | None
+    _set_fan_speed_topic: str | None
+    _state_topics: dict[str, Any]
+    _templates: dict[str, Callable[..., ReceivePayloadType]]
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        config: ConfigType,
+        config_entry: ConfigEntry,
+        discovery_data: DiscoveryInfoType | None,
+    ) -> None:
         """Initialize the vacuum."""
-        self._cleaning = False
-        self._charging = False
-        self._docked = False
-        self._error = None
-        self._status = "Unknown"
-        self._battery_level = 0
-        self._fan_speed = "unknown"
-        self._fan_speed_list = []
```

Comment:
Why was this initialization moved out of the constructor?

Reply:
was it?